mono/utils/mach-support-ppc.c

```diff
@@ -0,0 +1,139 @@
+/**
+ * \file
+ * mach support for PPC
+ *
+ * Authors:
+ *   Sergey Fedorov (vital.had@gmail.com)
+ *
+ * (C) 2023 Xamarin, Inc.
+ */
+
+/* The code below assumes 10.5.x or 10.6.x (10A190 or Rosetta).
```

Comment:
i'll note my previous half-hearted attempt was on 10.4, so this might explain why you had better success :)

Reply:
@NattyNarwhal Tiger is painful to develop for. I have a 10.4.11 set-up to run builds or test, but don’t really have motivation or time to work on it. 10.5.8 and 10A190/10.6.8 are reasonably homogeneous, on the other hand.

Reply:
(Registers are not prefixed on Tiger, so to support it here we need to add defines which would add `__` on >10.4, but not on <10.5. This is not hard, just extra code, but I have no idea if the rest of the code needs extra modifications.)